src/internal/shoelace-element.ts

```diff
@@ -92,6 +92,58 @@ export default class ShoelaceElement extends LitElement {
 
     return event as GetCustomEventType<T>;
   }
+
+  // @ts-expect-error
+  static version: string = typeof shoelaceVersion !== "undefined" ? shoelaceVersion : ""
+
+  static define (name: string) {
+    define(name, this)
+  }
+
+  static scopedElements: Record<string, typeof ShoelaceElement> = {}
+
+  connectedCallback () {
+    super.connectedCallback()
+    Object.entries((this.constructor as typeof ShoelaceElement).scopedElements).forEach(([name, component]) => {
+      define(name, component)
+    })
```

Comment:
Clever! So this means we no longer need the imports or `@dependency` tags, just the `scopedElements` property?

Reply:
Exactly. Key thing to note is to make sure to import "thing.component.js" so you don't accidentally register